docs/building.rst

```diff
@@ -647,3 +654,18 @@ in a command shell::
 
 This will also install the headers and library files in the visual studio directories to
 be picked up by libtorrent.
+
+list of macros
+--------------------
+
+The following is a list of defines that libtorrent is built with: ``BOOST_ALL_NO_LIB``, ``BOOST_ASIO_ENABLE_CANCELIO``, ``BOOST_ASIO_HAS_STD_CHRONO``, ``BOOST_MULTI_INDEX_DISABLE_SERIALIZATION``, ``BOOST_NO_DEPRECATED``, ``BOOST_SYSTEM_NO_DEPRECATED``, ``TORRENT_BUILDING_LIBRARY``, ``TORRENT_USE_I2P=1``
+
+Make sure you define the same at complile time for your code to avoid any runtime errors and other issues.
+
+These might change in the future, so it's always best to verify these every time you upgrade to
+a new version of libtorrent. The simplest way to see the full list of macros defined is to build
+libtorrent with "--debug-configuration -d2+2" switches added to b2 command line:
```

Comment:
Unless I'm misunderstanding what you're going for, I think `-n` is simpler. That will just print the command lines instead of executing them. If combined with `-a`, all command lines will be printed.

Reply:
Ok, I did not know that. Let's change to the way you are suggesting.